service/src/main/java/uk/gov/hmcts/reform/fpl/service/casesubmission/CaseSubmissionTemplateDataGenerationService.java

```diff
@@ -0,0 +1,741 @@
+    }
+
+    public void populateDocmosisCaseSubmissionWithCaseNumber(
+        final DocmosisCaseSubmission submittedCase, final long caseNumber) {
+        submittedCase.setCaseNumber(String.valueOf(caseNumber));
+    }
+
+    private String getApplicantsOrganisations(final List<Element<Applicant>> applicants) {
+        return applicants.stream()
+            .map(Element::getValue)
+            .filter(Objects::nonNull)
+            .map(Applicant::getParty)
+            .filter(Objects::nonNull)
+            .map(ApplicantParty::getOrganisationName)
+            .filter(StringUtils::isNotBlank)
+            .collect(joining(NEW_LINE));
+    }
+
+    private String getRespondentsNames(final List<Element<Respondent>> respondents) {
+        return respondents.stream()
+            .map(Element::getValue)
+            .filter(Objects::nonNull)
+            .map(Respondent::getParty)
+            .filter(Objects::nonNull)
+            .map(RespondentParty::getFullName)
+            .filter(StringUtils::isNotBlank)
+            .collect(joining(NEW_LINE));
+    }
+
+    private String getOrdersNeeded(final Orders orders) {
+        StringBuilder sb = new StringBuilder();
+
+        if (orders != null && isNotEmpty(orders.getOrderType())) {
+            sb.append(orders.getOrderType().stream()
+                .map(OrderType::getLabel)
+                .collect(joining(NEW_LINE)));
+
+            sb.append(NEW_LINE);
+            appendOtherOrderToOrdersNeeded(orders, sb);
+            appendEmergencyProtectionOrdersAndDetailsToOrdersNeeded(orders, sb);
+        }
+
+        return StringUtils.isNotEmpty(sb.toString()) ? sb.toString().trim() : DEFAULT_STRING;
+    }
+
+    private void appendOtherOrderToOrdersNeeded(final Orders orders, final StringBuilder stringBuilder) {
+        if (StringUtils.isNotEmpty(orders.getOtherOrder())) {
+            stringBuilder.append(orders.getOtherOrder());
+            stringBuilder.append(NEW_LINE);
+        }
+    }
+
+    private void appendEmergencyProtectionOrdersAndDetailsToOrdersNeeded(final Orders orders,
+                                                                         final StringBuilder stringBuilder) {
+        if (isNotEmpty(orders.getEmergencyProtectionOrders())) {
+            stringBuilder.append(orders.getEmergencyProtectionOrders().stream()
+                .map(EmergencyProtectionOrdersType::getLabel)
+                .collect(joining(NEW_LINE)));
+
+            stringBuilder.append(NEW_LINE);
+            if (StringUtils.isNotEmpty(orders.getEmergencyProtectionOrderDetails())) {
+                stringBuilder.append(orders.getEmergencyProtectionOrderDetails());
+            }
+        }
+    }
+
+    private String getDirectionsNeeded(final Orders orders) {
+        StringBuilder stringBuilder = new StringBuilder();
+        if (isNotEmpty(orders)
+            && (isNotEmpty(orders.getOrderType()) || StringUtils.isNotEmpty(orders.getDirections()))) {
```

Comment:
Maybe extract the condition. Something like `hasDirections()`. Also a bit confused as to why you are checking orders.getOrderType buy not actually using it.